api/app/controllers/spree/api/orders_controller.rb

```diff
@@ -27,7 +27,11 @@ class OrdersController < Spree::Api::BaseController
       def cancel
         authorize! :update, @order, params[:token]
         @order.canceled_by(current_api_user)
+
         respond_with(@order, default_template: :show)
+      rescue StateMachines::InvalidTransition => error
```

Comment:
With this addition, this controller includes these exact lines of codes 3 times... it's clearly becoming a pattern 😄  
I think it may be a good moment to DRY them up by extracting them in a private method... WDYT?

Reply:
Thank you for the comment @spaghetticode 
I think you mean [Api::CheckoutsController](https://github.com/solidusio/solidus/blob/master/api/app/controllers/spree/api/checkouts_controller.rb) where it is used two times. 

I agree that this could be a private method and that we can DRY. We could use a `rescue_from` in [Api::BaseController](https://github.com/solidusio/solidus/blob/master/api/app/controllers/spree/api/base_controller.rb#L30) to deal with `StateMachines::InvalidTransition` and return 422 in all api controllers.  

I'll move the rescue from  the api/orders and api/checkouts controllers to the api base controller if the above works. Let me know!

Reply:
@SamuelMartini oh yes, it's Api::CheckoutsController 😅 
I think your proposal is good 🎉